core/schema_utils.py

```diff
@@ -133,7 +135,20 @@ def normalize_against_schema(
                 prop[SCHEMA_KEY_SCHEMA],
                 global_validators=global_validators
             )
+    elif schema[SCHEMA_KEY_TYPE] == SCHEMA_TYPE_DICT_WITH_VARIABLE_NO_OF_KEYS:
+        assert isinstance(obj, dict), ('Expected dict, received %s' % obj)
+        schema_value_type = schema[SCHEMA_KEY_VALUES]
+        normalized_obj = {}
+        for key, value in obj.items():
+            assert isinstance(key, str), (
+                'Expected key to be string, received %s' % key)
```

Comment:
Could we make this more customizable? Basically, do smth like 
```python
normalized_key = normalize_against_schema(
    value, schema[SCHEMA_KEY_KEYS][SCHEMA_KEY_SCHEMA],
    global_validators=global_validators
)
normalized_obj[normalized_key] = normalize_against_schema(
    value, schema_value_type[SCHEMA_KEY_SCHEMA],
    global_validators=global_validators
)
```

Reply:
So the schema be something like this ?
``` js
mastery_change_per_skill: {
                'schema': {
                    'type': 'variable_keys_dict',
                    'key': {
                        'schema': {
                            'type':'basestring'
                        }
                    },
                     'values': {
                         'schema': {
                             'type': 'float'
                         }
                     }
                }
```

Reply:
Yes

Reply:
Done,